gsdk/src/signer/calls.rs

```diff
@@ -19,16 +19,27 @@
 //! gear api calls
 use crate::{
     config::GearConfig,
-    metadata::runtime_types::{gear_runtime::RuntimeCall, sp_weights::weight_v2::Weight},
+    metadata::runtime_types::{
+        frame_system::pallet::Call,
+        gear_common::{ActiveProgram, Program},
+        gear_core::code::InstrumentedCode,
+        gear_runtime::RuntimeCall,
+        sp_weights::weight_v2::Weight,
+    },
     signer::Signer,
-    types::{InBlock, TxStatus},
+    types::{self, InBlock, TxStatus},
+    Error,
 };
 use anyhow::anyhow;
 use async_recursion::async_recursion;
 use gear_core::ids::{CodeId, MessageId, ProgramId};
```

Comment:
Inconsistent usage of IDs throughout this crate confuses me:
- in some places it uses crate::metadata::gear_core::ids::{ProgramId, etc} whereas in this place it uses gear_core::{ProgramId, etc}
- at the same time the crate exposes some methods which use this similar abstractions from different crates (see comment below)
Why can't we use `gear_core::ids` everywhere? Why do we need `crate::metadata::gear_core::ids` at all? It causes a lot of confusing conversions.

Reply:
Right, I think we have to crate another issue and adjust the code so it makes use of `gear_core` as much as possible. Taking account of the fact this code is for our eco-system, the code has to be nice, clean and consistent as everything publicly facing

Reply:
Previously we had auto generated extrinsics and calls which were using metadata::ids::*
Due to the fact that we have now dynamic api, seems reasonable to use gear_core::ids everywhere (furthermore, publicly reimport it in gsdk/gclient maybe?)